lutris/sysoptions.py

```diff
@@ -99,77 +133,45 @@ def get_vk_icd_choices():
     amdvlk = []
     amdvlkpro = []
     icd_files = defaultdict(list)
+
+    vulkan_data_dirs = set()
+    for env_var in VULKAN_DATA_DIR_ENV_VARS:
+        env_var_val = os.getenv(env_var)
+        if env_var_val:
+            vulkan_data_dirs.update([os.path.join(dir, 'vulkan') for dir in env_var_val.split(':')])
+    vulkan_data_dirs.update(VULKAN_DATA_DIRS)
+
     # Add loaders
-    for data_dir in VULKAN_DATA_DIRS:
+    all_icd_files = set()
+
+    for data_dir in vulkan_data_dirs:
         path = os.path.join(data_dir, "icd.d", "*.json")
-        for loader in glob.glob(path):
-            icd_key = os.path.basename(loader).split(".")[0]
-            icd_files[icd_key].append(os.path.join(path, loader))
-            if "intel" in loader:
-                intel.append(loader)
-            elif "radeon" in loader:
-                amdradv.append(loader)
-            elif "nvidia" in loader:
-                nvidia.append(loader)
-            elif "amd" in loader:
-                if "pro" in loader:
-                    amdvlkpro.append(loader)
-                else:
-                    amdvlk.append(loader)
+        all_icd_files.update(glob.glob(path))
+
+    for loader in sorted(all_icd_files):
+        icd_key = os.path.basename(loader).split(".")[0]
+        icd_files[icd_key].append(os.path.join(path, loader))
+        if "intel" in loader:
+            intel.append(loader)
+        elif "radeon" in loader:
+            amdradv.append(loader)
+        elif "nvidia" in loader:
+            nvidia.append(loader)
+        elif "amd" in loader:
+            if "pro" in loader:
+                amdvlkpro.append(loader)
+            else:
+                amdvlk.append(loader)
 
     intel_files = ":".join(intel)
     amdradv_files = ":".join(amdradv)
     nvidia_files = ":".join(nvidia)
     amdvlk_files = ":".join(amdvlk)
     amdvlkpro_files = ":".join(amdvlkpro)
 
-    # Start the 'choices' with an 'auto' choice. But which one?
-    auto_intel_name = _("Auto: Intel Open Source (MESA: ANV)")
-    auto_amdradv_name = _("Auto: AMD RADV Open Source (MESA: RADV)")
-    auto_nvidia_name = _("Auto: Nvidia Proprietary")
-
-    vk_icd_filenames = os.getenv("VK_ICD_FILENAMES")
```

Comment:
This removes the VK_ICD_FILENAMES logic as the new logic is able to show what Auto will do if this is already set without any special casing - vulkaninfo will run with it set and show which GPU will be used.

It also allows non-auto options to override it even if it is set system-wide. If that's not desired I can add a check back and have all values default to "" if it's set.

In that case I'd also add `VK_DRIVER_FILES` and make `VK_ICD_FILENAMES` the fallback per the vulkan loader spec.

Reply:
If vulkaninfo is not available, then I think VK_ICD_FILENAMES will again be ignored, and issue #4610 will be back.

So, how common is it vulkaninfo?

Reply:
That particular issue wouldn't recur as Auto now leaves the env var unset, instead of trying to set it to whatever the default already was.

However if vulkaninfo isn't available and the system has multiple GPUs the text in the list showing which GPU corresponds with which ICD may be wrong.

Would it make more sense to not show which specific GPU corresponds to which ICD if vulkaninfo isn't available?

Reply:
It would be nice to know what is being selected, if that's possible.

But if vulkaninfo is universal anyway it may not really matter.